extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -451,29 +497,9 @@ export class PlatformStore {
 
   /**
    * Evaluates platforms and select the one to be selected for login.
-   * @return {!./subscription-platform.SubscriptionPlatform}
+   * @return {!Promise<!./subscription-platform.SubscriptionPlatform>}
    */
   selectPlatformForLogin() {
-    const platformScores = [];
-    this.getAvailablePlatforms().forEach(platform => {
-      let score = 0;
-      if (platform.supportsCurrentViewer()) {
-        score += 1000;
-      }
-      platformScores.push({
-        platform,
-        score,
-      });
-    });
-
-    platformScores.sort(function(platform1, platform2) {
-      return platform2.weight - platform1.weight;
-    });
-
-    if (platformScores.length === 0 || platformScores[0].score === 0) {
-      return this.getLocalPlatform();
-    }
-
-    return platformScores[0].platform;
+    return this.selectApplicablePlatform_('supportsViewer');
```

Comment:
Can we have a `@enum` of all factors in a separate file?

Reply:
SGTM will add

Reply:
done